packages/create/src/cli/add/change-data-source.ts

```diff
@@ -0,0 +1,146 @@
+
+let loginAttempts = 3;
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change-data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('change-data-source [path]')
+        .option('-s, --simulate', 'simulate only do not write or install')
+        .action(async (path, options) => {
+            await changeDataSource(path, { ...options }, !!options.simulate);
+        });
+}
+
+/**
+ * Changes the data source of an adaptation project.
+ *
+ * @param {string} basePath - The path to the adaptation project.
+ * @param {PromptDefaults} defaults - The default values for the prompts.
+ * @param {boolean} simulate - if set to true, then no files will be written to the filesystem
+ */
+async function changeDataSource(basePath: string, defaults: PromptDefaults, simulate: boolean): Promise<void> {
+    const logger = getLogger();
+    try {
+        if (!basePath) {
+            basePath = process.cwd();
+        }
+        checkEnvironment(basePath);
+
+        const variant = getVariant(basePath);
+        const manifest = await getManifest(basePath, defaults, logger, variant);
+        const dataSources = manifest['sap.app'].dataSources;
+        if (!dataSources) {
+            throw new Error('No data sources found in the manifest');
+        }
+        const answers = await promptYUIQuestions(getPromptsForChangeDataSource(dataSources), false);
+
+        const fs = await generateChange<ChangeType.CHANGE_DATA_SOURCE>(basePath, ChangeType.CHANGE_DATA_SOURCE, {
+            variant,
+            dataSources,
+            answers
+        });
+
+        if (!simulate) {
+            await new Promise((resolve) => fs.commit(resolve));
+        } else {
+            await traceChanges(fs);
+        }
+    } catch (error) {
+        logger.error(error.message);
+        if (error.code === 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' && !defaults.ignoreCertErrors) {
+            logger.error('If you are using a self-signed certificate, please use the --ignore-cert-errors flag.');
+            const confirm = await prompts([
+                {
+                    type: 'confirm',
+                    name: 'ignoreCertErrors',
+                    message: 'Do you want to ignore certificate errors?'
+                }
+            ]);
+            defaults.ignoreCertErrors = confirm.ignoreCertErrors;
+            await changeDataSource(basePath, defaults, simulate);
+            return;
+        }
+        if (error.response?.status === 401 && loginAttempts) {
+            loginAttempts--;
+            logger.error(`Authentication failed. Please check your credentials. Login attempts left: ${loginAttempts}`);
+            await changeDataSource(basePath, defaults, simulate);
+            return;
+        }
+        logger.debug(error);
+    }
+}
+
+/**
+ *
```

Comment:
Missing JSDoc description and returns

Reply:
Fixed with: cdd8157df89ff219fe223b367ae69befd7b86d13